controllers/backplaneconfig_controller.go

```diff
@@ -317,6 +326,27 @@ func (r *MultiClusterEngineReconciler) ensureCustomResources(ctx context.Context
 		return result, err
 	}
 
+	addonWorkManagerPath := "pkg/templates/core/clustermanageraddon_workmanager.yaml"
+	addonCRD := &apixv1.CustomResourceDefinition{}
+
+	err = r.Client.Get(ctx, types.NamespacedName{Name: clusterManagementAddonCRDName}, addonCRD)
+	if err != nil && apierrors.IsNotFound(err) {
+		log.Info(fmt.Sprintf("Waiting for CRD '%s' to be created by clustermanager resource", clusterManagementAddonCRDName))
+		return ctrl.Result{RequeueAfter: requeuePeriod}, nil
+	} else if err != nil {
+		return result, err
+	}
```

Comment:
To pull out details from the reconciler file this could be wrapped into a function and moved along with `getClusterManagementAddon` to `foundation/cluster_manager.go` or a new file. Then the logic and path details can be managed in the foundation package.

Reply:
I moved the reading of the files and the API check to the foundation package, but kept the apply in the controller to be more readable.